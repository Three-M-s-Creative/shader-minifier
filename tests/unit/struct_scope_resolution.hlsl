struct MyStructT{
  float f;
  static MyStructT Create(float f_in)
  {
    MyStructT my_struct;
    my_struct.f = f_in * 10;
    return my_struct;
  }
};

float Foo(float x)
{
  MyStructT my_struct = MyStructT::Create(x);
  return my_struct.f;
}