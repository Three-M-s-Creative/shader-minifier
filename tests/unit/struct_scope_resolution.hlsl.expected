struct N{
  float t;
  static N L(float M)
  {
    N r;
    r.t=M*10;
    return r;
  };
};
float r(float r)
{
  N M=N::L(r);
  return M.t;
}
